packages/address/src/utilities.ts

```diff
@@ -0,0 +1,74 @@
+import {Address, FieldName, Country, Province} from './types';
+
+const FIELD_REGEXP = /({\w+})/g;
+export const FIELDS_MAPPING: {
+  [key: string]: FieldName;
+} = {
+  '{firstName}': FieldName.FirstName, // eslint-disable-line id-length
+  '{lastName}': FieldName.LastName, // eslint-disable-line id-length
+  '{country}': FieldName.Country, // eslint-disable-line id-length
+  '{city}': FieldName.City, // eslint-disable-line id-length
+  '{zip}': FieldName.Zip, // eslint-disable-line id-length
+  '{province}': FieldName.Province, // eslint-disable-line id-length
+  '{address1}': FieldName.Address1, // eslint-disable-line id-length
+  '{address2}': FieldName.Address2, // eslint-disable-line id-length
+  '{phone}': FieldName.Phone, // eslint-disable-line id-length
+  '{company}': FieldName.Company, // eslint-disable-line id-length
+};
+
+/*
+ * Given a line template (eg. "{city} {zip}, {province}")
+ * Returns a string with the correct replacement
+ * If all replacement fields are empty, an empty string will be returned
+ */
+export function renderLineTemplate(
```

Comment:
I'm not sure how I feel about the inline documentation. cc @TheMallen. My thoughts are that we should do some combination of:
1. clarifying the names of the functions to more accurately describe what they do
2. adding documentation for any user-facing functions tot he readme file.

Reply:
I don't actually think this comment is helpful, the typescript code is pretty self explanatory. We can see just from the types that it's got the interface

```typescript
function renderLineTemplate(country: Country, template: string,  address: Address): string
```
The only non-obvious thing is that the return value is an empty string when the replacements are empty.

All that said, in general I'd be open to discussing autogenerating real docs from tsdoc comments in quilt libraries in the future if we thought it was really helpful.